From: Richard Acayan <mailingradian@gmail.com>
Date: Tue, 28 Jan 2025 22:58:36 -0500
Subject: [PATCH] arm64: dts: qcom: sdm670-google: add battery thermal zones

Battery charging is limited to 1 Amp to compensate for the lack of
working passive cooling. Add the thermal zones so faster charging can be
done safely.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi | 74 ++++++++++++++++++++++
 1 file changed, 74 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi b/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
index 8352e89..dca6e1c 100644
--- a/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
@@ -362,6 +362,72 @@ xo_crit: trip-point2 {
 				};
 			};
 		};
+
+		battery-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+
+			thermal-sensors = <&pm660_adc_tm 1>;
+
+			cooling-maps {
+				map0 {
+					trip = <&battery_alert0>;
+					cooling-device = <&pm660_charger 1 1>;
+				};
+
+				map1 {
+					trip = <&battery_alert1>;
+					cooling-device = <&pm660_charger 2 2>;
+				};
+
+				map2 {
+					trip = <&battery_alert2>;
+					cooling-device = <&pm660_charger 3 3>;
+				};
+
+				map3 {
+					trip = <&battery_alert3>;
+					cooling-device = <&pm660_charger 4 4>;
+				};
+
+				map4 {
+					trip = <&battery_alert4>;
+					cooling-device = <&pm660_charger 5 5>;
+				};
+			};
+
+			trips {
+				battery_alert0: trip-point0 {
+					temperature = <37500>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				battery_alert1: trip-point1 {
+					temperature = <39500>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				battery_alert2: trip-point2 {
+					temperature = <41500>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				battery_alert3: trip-point3 {
+					temperature = <42500>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				battery_alert4: trip-point4 {
+					temperature = <44500>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
 	};
 
 	/*
@@ -1036,6 +1102,14 @@ xo-therm@0 {
 		qcom,pre-scaling = <1 1>;
 		qcom,ratiometric;
 	};
+
+	msm-therm@1 {
+		reg = <1>;
+		io-channels = <&pm660_adc ADC5_AMUX_THM1_100K_PU>;
+		qcom,hw-settle-time-us = <200>;
+		qcom,pre-scaling = <1 1>;
+		qcom,ratiometric;
+	};
 };
 
 &pm660_charger {
